src/contexts/states/viewStore.ts

```diff
@@ -0,0 +1,208 @@
+import {create} from "zustand";
+
+import {Nullable} from "../../typings/common";
+import {
+    LOG_LEVEL,
+    LogLevelFilter,
+} from "../../typings/logs";
+import {DO_NOT_TIMEOUT_VALUE} from "../../typings/notifications";
+import {UI_STATE} from "../../typings/states";
+import {
+    BeginLineNumToLogEventNumMap,
+    CURSOR_CODE,
+    CursorType,
+    EVENT_POSITION_ON_PAGE,
+    PageData,
+} from "../../typings/worker";
+import {
+    ACTION_NAME,
+    NavigationAction,
+} from "../../utils/actions";
+import {clamp} from "../../utils/math";
+import {updateWindowUrlHashParams} from "../UrlContextProvider";
+import useContextStore from "./contextStore";
+import useLogFileManagerStore from "./LogFileManagerStore";
+import useLogFileStore from "./logFileStore";
+import useQueryStore from "./queryStore";
+import useUiStore from "./uiStore";
+
+
+interface ViewStoreValues {
+    beginLineNumToLogEventNum: BeginLineNumToLogEventNumMap;
+    logData: string;
+    numPages: number;
+    pageNum: number;
+}
+
+interface ViewStoreActions {
+    setBeginLineNumToLogEventNum: (newMap: BeginLineNumToLogEventNumMap) => void;
+    setLogData: (newLogData: string) => void;
+    setNumPages: (newNumPages: number) => void;
+    setPageNum: (newPageNum: number) => void;
+
+    updatePageData: (pageData: PageData) => void;
+    loadPageByAction: (navAction: NavigationAction) => void;
+    filterLogs: (filter: LogLevelFilter) => void;
+}
+
+type ViewState = ViewStoreValues & ViewStoreActions;
+
+const VIEW_STORE_DEFAULT: ViewStoreValues = {
+    beginLineNumToLogEventNum: new Map<number, number>(),
+    logData: "Loading...",
```

Comment:
We previously discussed here https://github.com/y-scope/yscope-log-viewer/pull/74#issuecomment-2359427747 
>> When no file is open, can we hide the icons and perhaps display something like "No file is open."
>> I know in a future PR, we'll probably hide this UI entirely when no file is open, but I think it's a small change to make for now?
>
>The initial fileName and also logData are now changed to "No file is open.".

We want to avoid showing a loading message when no log file is loaded to avoid confusions.
```suggestion
    logData: "No file is open.",
```